Marlin/src/gcode/lcd/M0_M1.cpp

```diff
@@ -85,7 +85,7 @@ void GcodeSuite::M0_M1() {
 
   #endif
 
-  TERN_(HOST_PROMPT_SUPPORT, hostui.prompt_do(PROMPT_USER_CONTINUE, parser.codenum ? F("M1 Stop") : F("M0 Stop"), FPSTR(CONTINUE_STR)));
+  TERN_(HOST_PROMPT_SUPPORT, hostui.prompt_do(PROMPT_USER_CONTINUE, FPSTR(parser.string_arg), FPSTR(CONTINUE_STR)));
```

Comment:
An SRAM-string can't be coerced to a flash string like this: `FPSTR(parser.string_arg)` and there is no facility to send a user string directly to `HostUI::prompt_do` – so the string sent to `HostUI::prompt_do` must be a wrapped constant `PROGMEM` string - i.e. an `F(...)` string.